app/models/school.rb

```diff
@@ -177,6 +177,8 @@ class ProcessDataError < StandardError; end
 
   validates_associated :school_times, on: :school_time_update
 
+  validates :alternative_heating_oil_percent, :alternative_heating_lpg_percent, :alternative_heating_biomass_percent, :alternative_heating_district_heating_percent, numericality: { greater_than_or_equal_to: 0, less_than_or_equal_to: 100, allow_blank: true }
```

Comment:
rails 6.1 has support for `validates ..., numericality: { in: 0..100 }` https://github.com/rails/rails/pull/41022